tests/GroupTest.php

```
@@ -41,6 +42,15 @@ public function testAddMiddleware(): void
         $this->assertSame($middleware2, $group->getData('enabledMiddlewares')[1]);
     }
 
+    public function testInvalidMiddlewares(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Invalid $middlewareDefinitions provided, list of string or array or callable expected.');
+
+        $middleware = static fn() => new Response();
+        $group = new Group('/api', [$middleware, new stdClass()]);
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$group'. (undefined)

(UnusedLocalVariable)